packages/svelte/src/reactivity/date.js

```diff
@@ -1,102 +1,132 @@
+/**
+ * @import { Source } from "#client"
+ */
+import { derived } from '../internal/client/index.js';
 import { source, set } from '../internal/client/reactivity/sources.js';
 import { get } from '../internal/client/runtime.js';
 
-/** @type {Array<keyof Date>} */
-const read = [
-	'getDate',
-	'getDay',
-	'getFullYear',
-	'getHours',
-	'getMilliseconds',
-	'getMinutes',
-	'getMonth',
-	'getSeconds',
-	'getTime',
-	'getTimezoneOffset',
-	'getUTCDate',
-	'getUTCDay',
-	'getUTCFullYear',
-	'getUTCHours',
-	'getUTCMilliseconds',
-	'getUTCMinutes',
-	'getUTCMonth',
-	'getUTCSeconds',
-	// @ts-expect-error this is deprecated
-	'getYear',
-	'toDateString',
-	'toISOString',
-	'toJSON',
-	'toLocaleDateString',
-	'toLocaleString',
-	'toLocaleTimeString',
-	'toString',
-	'toTimeString',
-	'toUTCString'
-];
-
-/** @type {Array<keyof Date>} */
-const write = [
-	'setDate',
-	'setFullYear',
-	'setHours',
-	'setMilliseconds',
-	'setMinutes',
-	'setMonth',
-	'setSeconds',
-	'setTime',
-	'setUTCDate',
-	'setUTCFullYear',
-	'setUTCHours',
-	'setUTCMilliseconds',
-	'setUTCMinutes',
-	'setUTCMonth',
-	'setUTCSeconds',
-	// @ts-expect-error this is deprecated
-	'setYear'
-];
+/**
+ * @typedef {Exclude<{[Key in keyof Date]: Date[Key] extends Function ? Key : never}[keyof Date], symbol>} DateMethodNames
+ */
 
 var inited = false;
-
 export class ReactiveDate extends Date {
+	/**
+	 * @type {Source<number>}
+	 */
 	#raw_time = source(super.getTime());
 
-	// We init as part of the first instance so that we can treeshake this class
+	/**
+	 * @type {Map<DateMethodNames, Source<unknown>>}
+	 */
+	#deriveds = new Map();
+
+	/**
+	 * @param {any[]} params
+	 */
+	constructor(...params) {
+		// @ts-ignore
+		super(...params);
+		this.#init();
+	}
+
 	#init() {
-		if (!inited) {
-			inited = true;
-			const proto = ReactiveDate.prototype;
-			const date_proto = Date.prototype;
+		if (inited) {
+			return;
+		}
+
+		inited = true;
+		var reactive_date_proto = ReactiveDate.prototype;
+		var date_proto = Date.prototype;
 
-			for (const method of read) {
+		/**
+		 * @type {Array<DateMethodNames>}
+		 */
+		var versioned_read_props = [
+			'getTimezoneOffset',
+			'getTime',
+			'toISOString',
+			'toLocaleString',
+			'toJSON',
+			'toUTCString',
+			'toString'
+		];
+
+		var fine_grained_read_props = /** @type {Array<DateMethodNames>} */ (
+			Object.getOwnPropertyNames(Date.prototype).filter(
+				(prop) =>
+					(prop.startsWith('get') || prop.startsWith('to')) &&
+					!versioned_read_props.includes(/** @type {DateMethodNames} */ (prop))
+			)
+		);
+
+		var write_props = /** @type {Array<DateMethodNames>} */ (
+			Object.getOwnPropertyNames(Date.prototype).filter((prop) => prop.startsWith('set'))
+		);
+
+		for (const method of fine_grained_read_props) {
+			// @ts-ignore
+			reactive_date_proto[method] = function (...args) {
+				// methods like getMinutes and getUTCMinutes can use the same signal
```

Comment:
I wouldn't worry about sharing the same derived for these cases:
1) it adds quite a bit more code and complexity
2) it's rare that someone will use both together

Reply:
Almost every getX has a getUTCX counterpart (also getX and getLocalX). This in theory, will create half amount of signals. Although I agree that its rare that someone will use both, yet again, I thought creating the least amount of signals is something that should be considered if possible. Should I remove it?

Reply:
Let’s remove it. No one will use both variants together and if they do it’s unobservable in performance